Extensions/DraggableBehavior/tests/draggableruntimebehavior.spec.js

```diff
@@ -51,112 +51,211 @@ describe('gdjs.DraggableRuntimeBehavior', function () {
+        object2.setPosition(450, 500);
+        if (firstInFront) {
+          object.setZOrder(2);
+          object2.setZOrder(1);
+        } else {
+          object.setZOrder(1);
+          object2.setZOrder(2);
+        }
+
+        // Drag'n'drop
+        runtimeScene.renderAndStep(1000 / 60);
+        runtimeGame.getInputManager().onMouseMove(450, 500);
+        runtimeGame
+          .getInputManager()
+          .onMouseButtonPressed(gdjs.InputManager.MOUSE_LEFT_BUTTON);
+        runtimeScene.renderAndStep(1000 / 60);
+        runtimeGame.getInputManager().onMouseMove(750, 600);
+        runtimeScene.renderAndStep(1000 / 60);
+        runtimeGame
+          .getInputManager()
+          .onMouseButtonReleased(gdjs.InputManager.MOUSE_LEFT_BUTTON);
+        runtimeScene.renderAndStep(1000 / 60);
+
+        if (firstInFront) {
+          // The 1st object moved
+          expect(object.getX()).to.be(750);
+          expect(object.getY()).to.be(600);
+          expect(object2.getX()).to.be(450);
+          expect(object2.getY()).to.be(500);
+        } else {
+          // The 2nd object moved
+          expect(object.getX()).to.be(450);
+          expect(object.getY()).to.be(500);
+          expect(object2.getX()).to.be(750);
+          expect(object2.getY()).to.be(600);
+        }
+      });
+    });
   });
-  it('should handle multitouch', function () {
-    runtimeGame.getInputManager().touchSimulateMouse(false);
-    object.setPosition(450, 500);
-    object2.setPosition(650, 600);
-
-    //Drag'n'drop
-    runtimeScene.renderAndStep(1000 / 60);
-    runtimeGame.getInputManager().onTouchStart(2, 450, 500);
-    runtimeGame.getInputManager().onTouchStart(1, 650, 600);
-    runtimeScene.renderAndStep(1000 / 60);
-    runtimeGame.getInputManager().onFrameEnded();
-    runtimeGame.getInputManager().onTouchMove(2, 750, 700);
-    runtimeGame.getInputManager().onTouchMove(1, 100, 200);
-    runtimeScene.renderAndStep(1000 / 60);
-    runtimeGame.getInputManager().onFrameEnded();
-    runtimeGame.getInputManager().onTouchEnd(2);
-
-    expect(object.getX()).to.be(750);
-    expect(object.getY()).to.be(700);
-    expect(object2.getX()).to.be(100);
-    expect(object2.getY()).to.be(200);
+  describe('(touch)', function () {
+    it('can drag an object', function () {
+      runtimeGame.getInputManager().touchSimulateMouse(false);
+      object.setPosition(450, 500);
+
+      // Drag'n'drop
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onTouchStart(1, 10, 20);
+      runtimeGame.getInputManager().onTouchStart(0, 450, 500);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
+      runtimeGame.getInputManager().onTouchMove(0, 750, 600);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
+      runtimeGame.getInputManager().onTouchEnd(0);
+      runtimeGame.getInputManager().onTouchEnd(1);
```

Comment:
Why finishing the touch #1? There are `onTouchMove(1, ` later that are called for this pointer and this seems to break the guarantee that input manager will have on touch start -> move > end called in this order.

Reply:
This was a mistake, I removed it.